include/api.php

```diff
@@ -1240,105 +1237,64 @@ function api_media_upload()
+ * @return string
+ * @throws Exception
  */
-function api_status_show($type, $item_id = 0)
+function api_status_show($type, $item_id)
 {
-	$a = \get_app();
-
-	$user_info = api_get_user($a);
-
-	Logger::log('api_status_show: user_info: '.print_r($user_info, true), Logger::DEBUG);
+	Logger::info(API_LOG_PREFIX . 'Start', ['action' => 'status_show', 'type' => $type, 'item_id' => $item_id]);
 
-	if (!empty($item_id)) {
-		// Get the item with the given id
-		$condition = ['id' => $item_id];
-	} else {
-		// get last public wall message
-		$condition = ['owner-id' => $user_info['pid'], 'uid' => api_user(),
-			'gravity' => [GRAVITY_PARENT, GRAVITY_COMMENT]];
-	}
+	$status_info = [];
 
-	if ($type == "raw") {
-		$condition['private'] = false;
+	$item = api_get_item(['id' => $item_id]);
+	if ($item) {
+		$status_info = api_format_item($item, $type);
 	}
 
-	$lastwall = Item::selectFirst(Item::ITEM_FIELDLIST, $condition, ['order' => ['id' => true]]);
-
-	if (DBA::isResult($lastwall)) {
-		$in_reply_to = api_in_reply_to($lastwall);
-
-		$converted = api_convert_item($lastwall);
-
-		if ($type == "xml") {
-			$geo = "georss:point";
-		} else {
-			$geo = "geo";
-		}
+	Logger::info(API_LOG_PREFIX . 'End', ['action' => 'get_status', 'status_info' => $status_info]);
 
-		$status_info = [
-			'created_at' => api_date($lastwall['created']),
-			'id' => intval($lastwall['id']),
-			'id_str' => (string) $lastwall['id'],
-			'text' => $converted["text"],
-			'source' => (($lastwall['app']) ? $lastwall['app'] : 'web'),
-			'truncated' => false,
-			'in_reply_to_status_id' => $in_reply_to['status_id'],
-			'in_reply_to_status_id_str' => $in_reply_to['status_id_str'],
-			'in_reply_to_user_id' => $in_reply_to['user_id'],
-			'in_reply_to_user_id_str' => $in_reply_to['user_id_str'],
-			'in_reply_to_screen_name' => $in_reply_to['screen_name'],
-			'user' => $user_info,
-			$geo => null,
-			'coordinates' => '',
-			'place' => '',
-			'contributors' => '',
-			'is_quote_status' => false,
-			'retweet_count' => 0,
-			'favorite_count' => 0,
-			'favorited' => $lastwall['starred'] ? true : false,
-			'retweeted' => false,
-			'possibly_sensitive' => false,
-			'lang' => '',
-			'statusnet_html' => $converted["html"],
-			'statusnet_conversation_id' => $lastwall['parent'],
-			'external_url' => System::baseUrl() . '/display/' . $lastwall['guid'],
-		];
-
-		if (count($converted["attachments"]) > 0) {
-			$status_info["attachments"] = $converted["attachments"];
-		}
+	return api_format_data('statuses', $type, ['status' => $status_info]);
+}
 
-		if (count($converted["entities"]) > 0) {
-			$status_info["entities"] = $converted["entities"];
-		}
+/**
+ * Retrieves the last public status of the provided user info
+ *
+ * @param int    $ownerId Public contact Id
+ * @param int    $uid     User Id
+ * @param string $type    Return format (atom, rss, xml, json)
+ * @return array
+ * @throws Exception
+ */
+function api_get_last_status($ownerId, $uid, $type = 'json')
```

Comment:
I'd like a test at `ApiTest` because this is a new function and as @rudloff said he tried to add at least one test for each function in API (which should help us in a later state of transferring API to classes).

A guess (at `tests/include/ApiTest.php`):
```php
	/**
	 * Test the api get_last_status() function
	 */
	public function testApiGetLastStatus()
	{
		$result = api_get_last_status($this->friendUser['id'], $this->selfUSer['id'], 'json');
		$this->assertNotNull($result);
		...
	}
```
```
All data, which are available during a test run can found here:
https://github.com/friendica/friendica/blob/develop/tests/datasets/api.yml

So you can access 6 different items and three contacts, which are also defined at `setUp()` of `ApiTest.php`